src/drivers/imx/sai.c

```diff
@@ -176,25 +176,29 @@ static inline int sai_set_config(struct dai *dai,
 	/* DAI clock master masks */
 	switch (config->format & SOF_DAI_FMT_MASTER_MASK) {
 	case SOF_DAI_FMT_CBS_CFS:
-		break;
-	case SOF_DAI_FMT_CBM_CFM:
+		trace_sai("SAI: codec is slave");
 		val_cr2 |= REG_SAI_CR2_MSEL_MCLK1;
 		val_cr2 |= REG_SAI_CR2_BCD_MSTR;
+		val_cr2 |= SAI_CLOCK_DIV; // TODO: determine dynamically.
 		val_cr4 |= REG_SAI_CR4_FSD_MSTR;
 		break;
+	case SOF_DAI_FMT_CBM_CFM:
+		trace_sai("SAI: codec is master");
+		// fields CR2_DIV and CR2_MSEL not relevant in slave mode.
+		// fields CR2_BCD and CR4_MFSD already at 0
```

Comment:
Please use C style comments like this:
/*
 * xyz
 */

```

Reply:
Thanks Daniel